lib/base/custom_layers/pbf_layer/weather/weather_marker_modal.dart

```diff
@@ -9,11 +9,61 @@ import 'package:trufi_core/base/models/trufi_place.dart';
 class ParkingMarkerModal extends StatelessWidget {
   final WeatherFeature parkingFeature;
   final void Function() onFetchPlan;
+
   const ParkingMarkerModal({
     Key? key,
     required this.parkingFeature,
     required this.onFetchPlan,
   }) : super(key: key);
+
+  String _mapRoadCondition(int? value, bool isEnglish) {
+    if (value == null) return isEnglish ? "Unknown" : "Unbekannt";
+    switch (value) {
+      case 10:
+        return isEnglish ? "Dry" : "Trocken";
+      case 15:
+        return isEnglish ? "Humid" : "Feucht";
+      case 20:
+        return isEnglish ? "Wet" : "Nass";
+      case 25:
+        return isEnglish ? "Humid with salt" : "Feucht mit Salz";
+      case 30:
+        return isEnglish ? "Wet with salt" : "Nass mit Salz";
+      case 35:
+        return isEnglish ? "Ice" : "Eis";
+      case 40:
+        return isEnglish ? "Snow" : "Schnee";
+      case 45:
+        return isEnglish ? "Frost" : "Frost / Reif";
+      default:
+        return isEnglish ? "Unknown" : "Unbekannt";
+    }
+  }
+
+  String _mapPrecipitationType(int? value, bool isEnglish) {
+    if (value == null) return isEnglish ? "Unknown" : "Unbekannt";
+    switch (value) {
+      case 0:
+        return isEnglish ? "No precipitation" : "kein Niederschlag";
+      case 60:
+        return isEnglish
+            ? "Liquid precipitation, e.g. rain"
+            : "flüssiger Niederschlag, z.B. Regen";
+      case 70:
+        return isEnglish
+            ? "Solid precipitation, e.g. snow"
+            : "fester Niederschlag, z.B. Schnee";
+      case 67:
+        return isEnglish ? "Freezing rain" : "Eisregen";
+      case 69:
+        return isEnglish ? "Sleet" : "Schneeregen";
+      case 90:
+        return isEnglish ? "Hail" : "Hagel";
+      default:
+        return isEnglish ? "Unknown" : "Unbekannt";
+    }
+  }
```

Comment:
Is this the way you usually handle I18N in the Trufi App? I'd prefer to see user facing strings in a translation file so all messages are in one place and might be translated to another locale, if intended.

Reply:
Hi @hbruch this is the "custom" way we currently handle the languag for this feature. As mentioned we are currently working on streamlining / improving the code. This will also potentially be a part of the process. Thank you for the notice!